src/uri.jl

```diff
@@ -145,13 +145,13 @@ function Base.isvalid(uri::URI)
 end
 
 lower(c::UInt8) = c | 0x20
-@inline shouldencode(c) = c < 0x1f || c > 0x7f || @anyeq(c,
-                            UInt8(';'), UInt8('/'), UInt8('?'), UInt8(':'),
-                            UInt8('@'), UInt8('='), UInt8('&'), UInt8(' '),
-                            UInt8('"'), UInt8('<'), UInt8('>'), UInt8('#'),
-                            UInt8('%'), UInt8('{'), UInt8('}'), UInt8('|'),
-                            UInt8('\\'), UInt8('^'), UInt8('~'), UInt8('['),
-                            UInt8(']'), UInt8('`'))
+# RFC3986 Unreserved Characters (and '~' Unsafe per RFC1738).
+@inline shouldencode(c) = !((c >= UInt8('A') && c <= UInt8('Z'))
+                         || (c >= UInt8('a') && c <= UInt8('z'))
+                         || (c >= UInt8('0') && c <= UInt8('9'))
+                         || c == UInt8('-')
+                         || c == UInt8('.')
+                         || c == UInt8('_'))
```

Comment:
So AWS is more permissive in what they allow in their query parameter keys/values? The link you shared was a bit unwieldy; are we just encoding too much right now?